### PR TITLE
Merge earth and ocean polygons

### PR DESCRIPTION
**Pull request by iandees**
- [x] Update tests
- [ ] ~~Update data migrations~~
- [ ] ~~Update docs~~

Connects to #1106.

### CODE REVIEW

File: integration-test/1106-merge-ocean-earth.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# There should be a single (merged) ocean feature in this tile
+assert_less_than_n_features(9, 167, 186, 'ocean', properties, 2)
+# There should be a single (merged) earth feature in this tile
+assert_less_than_n_features(9, 170, 186, 'earth', properties, 2)
```
Comment (nvkelso) [inline integration-test/1106-merge-ocean-earth.py]: We should test once for OSM and once again for Natural Earh, so these tests again but for a zoom 5 coordinate.

Comment (iandees) [inline integration-test/1106-merge-ocean-earth.py]: Yep, I was starting here while I started figuring out how to replicate the tiled Natural Earth we use.

File: integration-test.py [hunk 2]
```diff
@@ -311,6 +311,25 @@ def assert_at_least_n_features(z, x, y, layer, properties, n):
...
+def assert_less_than_n_features(z, x, y, layer, properties, n):
+    """
+    Downloads a tile and checks that it contains less than `n` features which
+    match the given `properties`.
+    """
+    with features_in_tile_layer(z, x, y, layer) as features:
+        num_features, num_matching = count_matching(
+            features, properties)
+
+        if num_features >= n:
```
Comment (zerebubuth) [inline integration-test.py]: I'm not sure that this fits with the description of the function.

For example, if a tile has 10 features, but only one matches the `properties`, and we test `assert_less_than_n_features(z, x, y, layer, properties, 2)`, then I would expect the test to pass. But the test on the total number of features here means that it won't.

File: integration-test.py [hunk 3]
```diff
@@ -359,6 +378,8 @@ def print_coords(f, log, idx, num_tests):
         runpy.run_path(f, init_globals={
             'assert_has_feature': print_coord,
             'assert_no_matching_feature': print_coord,
+            'assert_at_least_n_features': assert_at_least_n_features,
+            'assert_less_than_n_features': assert_less_than_n_features,
```
Comment (zerebubuth) [inline integration-test.py]: These should be `print_coord`, or similar function with matching signature. This part of the code is used to print the coordinates of all tiles being tested, which can be useful for pre-seeding the dev TOI.

File: integration-test.py (hunk 2 above)
Comment (iandees) [inline integration-test.py]: I'm just going to remove the `num_features` check altogether and rely on the `num_matching` check. Does that make sense?

Comment (zerebubuth) [inline integration-test.py]: Yup, that's good.

File: queries.yaml [hunk 4]
```diff
@@ -425,6 +425,15 @@ post_process:
...
         'label_placement' not in properties and kind in
         ('archipelago', 'island', 'islet', 'continent', 'valley', 'ridge')
+
+  # merge ocean and earth polygons, see https://github.com/tilezen/vector-datasource/issues/1106
+  - fn: vectordatasource.transform.merge_polygon_features
+    params:
+      source_layer: earth
+  - fn: vectordatasource.transform.merge_polygon_features
+    params:
+      source_layer: ocean
```
Comment (rmarianski) [inline queries.yaml]: Should this be `water` instead?

Comment (iandees) [inline queries.yaml]: Dangit, yes. I thought I had fixed that earlier.

**Comment by rmarianski** [comment]
@iandees btw, you get the :trophy: for best branch name of the day. In my mind, I say "heart" instead of leelu in reference to captain planet. And "multi-pass" when I hear leelu.

**Comment by iandees** [comment]
I'm not seeing any tiled/unjoined `kind:ocean` data at the < z9 range (the Natural Earth data), so a test there wouldn't really be testing anything new.

**Comment by iandees** [comment]
@nvkelso lowzoom tests added. You're thinking of screenshots from something like [the tag explorer map](http://tangrams.github.io/explorer/#10.2/43.8042/-69.0568/kind/ocean) before and after? How do I go about switching the tag explorer to use my local tileserver endpoint?

**Comment by nvkelso** [comment]
@iandees clone that repo, start up local http server and load the index file, point the YAML source at your dev ip & port.

**Comment by nvkelso** [comment]
You could also write quick Tangram Play style...

**Comment by nvkelso** [comment]
I think this is ready to merge, we can visually verify it once it's on dev.

**Pull request merged**